pilot/pkg/xds/debug.go

```diff
@@ -100,19 +100,20 @@ type AdsClients struct {
 	Connected []AdsClient `json:"clients"`
 }
 
+type PerXDSSyncStatus struct {
+	Type         string     `json:"type,omitempty"`
+	NonceSent    string     `json:"nonce_sent,omitempty"`
+	NonceAcked   string     `json:"nonce_acked,omitempty"`
+	LastSyncTime *time.Time `json:"last_sync_time,omitempty"`
+}
+
 // SyncStatus is the synchronization status between Pilot and a given Envoy
 type SyncStatus struct {
-	ProxyID       string `json:"proxy,omitempty"`
-	ProxyVersion  string `json:"proxy_version,omitempty"`
-	IstioVersion  string `json:"istio_version,omitempty"`
-	ClusterSent   string `json:"cluster_sent,omitempty"`
-	ClusterAcked  string `json:"cluster_acked,omitempty"`
-	ListenerSent  string `json:"listener_sent,omitempty"`
-	ListenerAcked string `json:"listener_acked,omitempty"`
-	RouteSent     string `json:"route_sent,omitempty"`
-	RouteAcked    string `json:"route_acked,omitempty"`
-	EndpointSent  string `json:"endpoint_sent,omitempty"`
-	EndpointAcked string `json:"endpoint_acked,omitempty"`
+	ProxyID      string `json:"proxy,omitempty"`
+	ProxyVersion string `json:"proxy_version,omitempty"`
+	IstioVersion string `json:"istio_version,omitempty"`
+
+	Statuses map[string]PerXDSSyncStatus `json:"statuses,omitempty"`
```

Comment:
Sorry I should have mentioned this originally, we may need to keep both the old and new style around for a while, as we need to be backwards compatible with older istioctl versions. is that accurate @esnible ?

Reply:
Ok. That should not be a problem. I'll add it back and mark it deprecated (in comments)

Reply:
Added it back, but marked those fields `Deprecated`.